tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -5952,25 +6079,96 @@ void ExecutionTest::RunRawBufferLdStTest(D3D_SHADER_MODEL shaderModel, RawBuffer
        WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
        return;
      }
+     sTy = (dataType == RawBufferLdStType::I16 ? "int16_t" : "half");
      break;
    case RawBufferLdStType::I32:
+     sTy = "int32_t";
+     break;
    case RawBufferLdStType::Float:
+     sTy = "float";
      break;
    default:
      DXASSERT_NOMSG("Invalid RawBufferLdStType");
    }
 
-   if (shaderOpName == nullptr) {
-     // TODO: finish up all variations
-     WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
-     return;
-   }
-
+   // format shader text
+   std::string shaderText(RawBufferTestComputeShaderTemplate);
+   ReplaceAll(shaderText, std::string("DATATYPE_SIZE"), std::to_string(sizeof(Ty)));
+   ReplaceAll(shaderText, std::string("DATATYPE"), sTy);
+ 
+   // read shader config
    CComPtr<IStream> pStream;
    ReadHlslDataIntoNewStream(L"ShaderOpArith.xml", &pStream);
 
-   // TODO: finish up
-   WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+   // define test data structures
+   struct TestData {
+     Ty v1, v2[2], v3[3], v4[4];
+   };
+
+   struct UavData {
+     TestData input, output, srvOut;
+   };
+
+   // run the shader
+   std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTest(pDevice, m_support, pStream, shaderOpName,
+     [&](LPCSTR Name, std::vector<BYTE> &Data, st::ShaderOp *pShaderOp) {
+     VERIFY_IS_TRUE(((0 == strncmp(Name, "SRVBuffer", 9)) || (0 == strncmp(Name, "UAVBuffer", 9))) &&
+                    (Name[9] >= '0' && Name[9] <= '3'));
+     pShaderOp->Shaders.at(0).Text = shaderText.data();
+     TestData *pInData = (TestData*)Data.data();
```

Comment:
I would VERIFY that this has the expected size before `memcpy'ing` into it.

Reply:
I Agree.  Assuming that this vector is initialized to the size of the buffer defined in the XML, any mismatch between these could manifest as a nasty bug.  Verifying the size is the best way to catch the root cause.